README.md

```diff
@@ -66,42 +66,122 @@ In order to use the SDK installed via vcpkg with CMake, you can use the toolchai
+#### Key Core concepts
+
+Understanding the key concepts from the `Azure Core` library, which is leveraged by all client libraries will also be helpful in getting started, regardless of which Azure service you want to use.
+
+The main shared concepts of `Azure Core` include:
+
+- Accessing HTTP response details for the returned model of any SDK client operation, via `Response<T>`.
+- Exceptions for reporting errors from service requests in a consistent fashion via the base exception type `RequestFailedException`.
+- Abstractions for Azure SDK credentials (`TokenCredential`).
+- Handling streaming data and input/output (I/O) via `BodyStream` along with its derived types.
+- Polling long-running operations (LROs), via `Operation<T>`.
+- Operations that return collections do so via a `PagedResponse<T>`.
+- HTTP pipeline and HTTP policies such as retry and logging, which are configurable via service client specific options.
+- Replaceable HTTP transport layer to send requests and receive responses over the network.
+
+#### `Response <T>` Model Types
+
+Many client library operations **return** the templated `Azure::Core::Response<T>` type from the API calls. This type let's you get the raw HTTP response from the service request call the Azure service APIs make, along with the result of the operation to get more API specific details. This is the templated `T` operation result which can be extracted from the response, using the `Value` field.
+
+```C++
+  // Azure service operations return a Response<T> templated type.
+  Azure::Response<Models::BlobProperties> propertiesResponse = blockBlobClient.GetProperties();
+
+  // You can get the T, from the returned Response<T>,
+  // which is typically named with a Result suffix in the type name.
+  Models::BlobProperties propertiesModel = propertiesResponse.Value;
+
+  // Now you can look at API specific members on the result object that is returned.
+  std::cout << "The size of the blob is: " << propertiesModel.BlobSize << std::endl;
+```
+
+#### Long Running Operations
+
+Some operations take a long time to complete and require polling for their status. Methods starting long-running operations return `Operation<T>` types.
+
+You can intermittently poll whether the operation has finished by using the `Poll()` method inside a loop on the returned `Operation<T>` and track progress of the operation using `Value()`, while the operation is not done (using `IsDone()`). Your per-polling custom logic can go in that loop, such as logging progress.Alternatively, if you just want to wait until the operation completes, you can use `PollUntilDone()`.
+
+```C++
+  std::string sourceUri = "<a uri to the source blob to copy>";

```

Comment:
nit: mismatch between url (above) and url (here) usage. Also, maybe capitalize?

Reply:
I know. The API is called StartCopyFromUri, so that's where the inconsistency issue comes from :(

Reply:
I see, the API is called ...FormUri, it is better to align with the API then.